quixstreams/kafka/admin.py

```diff
@@ -0,0 +1,195 @@
+        else:
+            logger.debug(f"Creating topics via Kafka API")
+            creator = self._create_topics
+        creator([topic.as_creation_config() for topic in topics])
+
+    def auto_create_or_validate_topics(self):
+        topics = self.topics + self.changelog_topics
+        topics = [topic.kafka_configs for topic in topics]
+        logger.info(f"topics required for app operation: {[t.name for t in topics]}")
+        if self._auto_create_topics:
+            logger.info(
+                "auto-creation is enabled: attempting to create required topics"
+            )
+            self.create_topics(topics)
+        else:
+            logger.info("auto-creation is disabled: validating required topics exist")
+            self.validate_topics(topics)
+
+    def validate_topics(self, topics: Union[List[Topic], List[TopicKafkaConfigs]]):
+        # TODO: add non-quix validator
+        if self.has_quix_builder:
+            self.quix_config_builder.confirm_topics_exist(topics)
+
+    def topic(
+        self,
+        name: str,
+        value_deserializer: Optional[DeserializerType] = None,
+        key_deserializer: Optional[DeserializerType] = "bytes",
+        value_serializer: Optional[SerializerType] = None,
+        key_serializer: Optional[SerializerType] = "bytes",
+        kafka_configs: Optional[TopicKafkaConfigs] = None,
+    ) -> Topic:
+        if self._auto_create_topics:
+            if self.has_quix_builder:
+                name = self.quix_config_builder.append_workspace_id(name)
+            if kafka_configs:
+                kafka_configs = TopicKafkaConfigs(
+                    name=name,
+                    num_partitions=kafka_configs.num_partitions,
+                    replication_factor=kafka_configs.replication_factor,
+                    optionals=kafka_configs.optionals,
+                    is_quix_topic=self.has_quix_builder,
+                )
+            else:
+                kafka_configs = TopicKafkaConfigs(
+                    name=name, is_quix_topic=self.has_quix_builder
+                )
+        topic = Topic(
+            name=name,
+            value_serializer=value_serializer,
+            value_deserializer=value_deserializer,
+            key_serializer=key_serializer,
+            key_deserializer=key_deserializer,
+            kafka_configs=kafka_configs,
+        )
+        self._topics[name] = topic
+        return topic
+
+    def changelog_topic(
+        self,
+        source_topic_name: str,
+        suffix: str,
+        configs_to_import: tuple = ("retention.bytes", "retention.ms"),
+    ) -> Topic:
+        topic_configs = (
+            self.inspect_topics([source_topic_name])[source_topic_name]
+            or self._topics[source_topic_name].kafka_configs
+        )
+        name = f"__changelog--{self._consumer_group}--{source_topic_name}--{suffix}"
+        base_optionals = {} if self.has_quix_builder else {"cleanup.policy": "compact"}
```

Comment:
I'd consider expanding "optionals" into the individual named parameters if it's possible.
(In other places too)

If not, let's rename it to `extra` like it's done in Producer and Consumer classes